bucket-cache.go

```diff
@@ -102,6 +102,7 @@ func (c Client) getBucketLocation(ctx context.Context, bucketName string) (strin
 		return "", err
 	}
 
+	req = req.WithContext(ctx)
 	// Initiate the request.
 	resp, err := c.do(req)
 	defer closeResponse(resp)
```

Comment:
it is better to use http.NewRequestWithContext() instead of req.WithContext @alexcocia

Reply:
Here is the change I would expect

```diff
git diff
diff --git a/bucket-cache.go b/bucket-cache.go
index 7d485a6..156150f 100644
--- a/bucket-cache.go
+++ b/bucket-cache.go
@@ -97,7 +97,7 @@ func (c Client) getBucketLocation(ctx context.Context, bucketName string) (strin
        }
 
        // Initialize a new request.
-       req, err := c.getBucketLocationRequest(bucketName)
+       req, err := c.getBucketLocationRequest(ctx, bucketName)
        if err != nil {
                return "", err
        }
@@ -169,7 +169,7 @@ func processBucketLocationResponse(resp *http.Response, bucketName string) (buck
 }
 
 // getBucketLocationRequest - Wrapper creates a new getBucketLocation request.
-func (c Client) getBucketLocationRequest(bucketName string) (*http.Request, error) {
+func (c Client) getBucketLocationRequest(ctx context.Context, bucketName string) (*http.Request, error) {
        // Set location query.
        urlValues := make(url.Values)
        urlValues.Set("location", "")
@@ -198,7 +198,7 @@ func (c Client) getBucketLocationRequest(bucketName string) (*http.Request, erro
        }
 
        // Get a new HTTP request for the method.
-       req, err := http.NewRequest(http.MethodGet, urlStr, nil)
+       req, err := http.NewRequestWithContext(ctx, http.MethodGet, urlStr, nil)
        if err != nil {
                return nil, err
        }
```

Reply:
New commit provided applying your code. Thanks a lot, that was a quick response!